modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/classloader/DefaultArtifactClassLoaderResolver.java

```diff
@@ -74,13 +76,22 @@ public class DefaultArtifactClassLoaderResolver implements ArtifactClassLoaderRe
   private final ModuleRepository moduleRepository;
   private final NativeLibraryFinderFactory nativeLibraryFinderFactory;
   private final MuleDeployableArtifactClassLoader defaultDomainClassloader;
+  private final FeatureFlaggingService featureFlaggingService;
 
   public DefaultArtifactClassLoaderResolver(ArtifactClassLoader containerClassLoader,
                                             ModuleRepository moduleRepository,
                                             NativeLibraryFinderFactory nativeLibraryFinderFactory) {
+    this(containerClassLoader, moduleRepository, nativeLibraryFinderFactory, null);
```

Comment:
I don't like the fact that constructing this instance with a `null` `FeatureFlaggingService` is still allowed, that could become a headache down the line.
Changing this would require changing the [mule-framework api](https://github.com/mulesoft/mule/blob/255a790e2a565d71113c181b30dce2b3aff25aff/modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/api/classloader/ArtifactClassLoaderResolver.java#L58). That wouldn't be too much of a problem because it is still not GA, but I don't know if we want `mule-framework` users to have to be aware about feature flagging.
Any thoughts?